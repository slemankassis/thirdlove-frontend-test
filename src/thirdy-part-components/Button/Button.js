import React from 'react';

class Button extends React.Component {
  render() {
    return (
      1
    );
  }
}

export default Button;
