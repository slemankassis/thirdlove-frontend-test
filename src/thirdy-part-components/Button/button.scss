.button {
  background-color: black;
  display: none;
}
