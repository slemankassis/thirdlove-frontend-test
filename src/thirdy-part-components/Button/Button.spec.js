import React from 'react';
import Button from './Button';

describe('Button snaps', () => {
  it('renders snapshot without props', () => {
    const component = shallow(<Button />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });

  it('renders snapshot with fullProps', () => {
    const fullProps = {
      text: '__TEXT__',
      className: '__CLASS_NAME__',
      onClick: jest.fn(),
    };

    const component = shallow(<Button {...fullProps} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });
});
