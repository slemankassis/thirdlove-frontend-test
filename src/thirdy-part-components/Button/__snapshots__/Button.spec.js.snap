// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders snapshot with fullProps 1`] = `
"<button className='__TEXT__ __CLASS_NAME__' type='button'>
  __TEXT__
</button>"
`;

exports[`Button renders snapshot without props 1`] = `"<button className='' type='button' />"`;
